src/scene_battle_rpg2k.cpp

```diff
@@ -418,26 +418,28 @@ bool Scene_Battle_Rpg2k::ProcessBattleAction(Game_BattleAlgorithm::AlgorithmBase
 
 			std::vector<int16_t> states_to_heal = action->GetSource()->NextBattleTurn();
 			std::vector<int16_t> states_remaining = action->GetSource()->GetInflictedStates();
+			std::string message_to_print;
 			action->GetSource()->ApplyConditions();
 			bool message_to_show = false;
 			if (!states_to_heal.empty() || !states_remaining.empty()) {
 				for (auto state_id : states_to_heal) {
 					// BattleAlgorithm verifies the states
 					const RPG::State* state = ReaderUtil::GetElement(Data::states, state_id);
 					if (!state->message_recovery.empty()) {
-						battle_message_window->PushWithSubject(state->message_recovery, action->GetSource()->GetName());
+						message_to_print = state->message_recovery;
 						message_to_show = true;
 					}
 				}
 				for (auto state_id : states_remaining) {
 					// BattleAlgorithm verifies the states
 					const RPG::State* state = ReaderUtil::GetElement(Data::states, state_id);
 					if (!state->message_affected.empty()) {
-						battle_message_window->PushWithSubject(state->message_affected, action->GetSource()->GetName());
+						message_to_print = state->message_affected;
```

Comment:
Need to test this. Might it be state priority instead of just the last one that dictates which message is shown?

Reply:
This one is wrong. Fixed in #1451